File: 2 semester/3 home work/2-3-4/2-3-4/Stack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _2_3_4
{
    public class Stack: IStack
    {
        private int amountOfElements = 0;
        private StackElement head = null;

        public void Push(float value)
        {
            StackElement tmp = new StackElement(head, value);
            head = tmp;
            amountOfElements++;
        }

        public float Pop()
        {
            if (head != null)
            {
                float tmp = head.Element;
                amountOfElements--;
                head = head.Next;
                return tmp;
            }
            else
            {
                throw new EmptyStackException("Empty Stack"); ;
            }
        }

        public int AmountOfElements()
        {
            return amountOfElements;
        }
    }
}

File: 2 semester/java/2 home work/2_2_1/src/pkg2_2_1/Stack.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package pkg2_2_1;

/**
 *
 * @author Alex
 */
public class Stack {
    private StackElement head;
    
    public Stack() {
        head = null;
    }
    
    public boolean isEmpty() {
        return this.head == null;
    }
    
    public void push(int element) {
        StackElement tmp = new StackElement(element, this.head);
        this.head = tmp;
    }
    
    public int pop() {
        if (this.isEmpty()) {
            return -999;
        } else {
            int tmp = this.head.element();
            this.head = this.head.next();
            return tmp;
        }
    }
    
    public void printStack() {
        StackElement iterator = this.head;
        while (iterator != null) {
            System.out.println(iterator.element());
            iterator = iterator.next();
        }
    }
}
